### DISCUSSION TITLE
How to apply contains check on the body in Rules

### DISCUSSION THREAD

**Discussion by ParveeenRao** [mockoon/mockoon#1204] [Help]
Hi, 

I have mocked a POST endpoint that receives requests in URL-encoded format. I need to create a rule based on the value of a field in the request or include a 'contains' condition in the rule for the request body. How I can achieve it ? 

Thanks,
Parveen

**Comment by 255kb** [comment]
I would need more details, with an exact example of the data you are trying to check the rules against (the entering body, etc). 
But while there is no array contains in the rules, nor string includes, you can normally use regexes to check if a value is present. 
Something like this: 
![image](https://github.com/mockoon/mockoon/assets/7489814/c1f387c6-06b2-4312-a6f2-3003f0b16178)

With the following entering body: 
```
{
  "arr": ["test1","test2"]
}
```

If you want to check if the body contains something, anywhere, you can leave the second field empty, and it will check in the full body stringified: 
![image](https://github.com/mockoon/mockoon/assets/7489814/098ab603-b706-477a-b823-060871b49d63)

Everything is documented [here](https://mockoon.com/docs/latest/route-responses/dynamic-rules/).

**Comment by ParveeenRao** [comment]
Hey , 

Thanks for your reply. It worked for me
